src/node_url.cc

```diff
@@ -1283,7 +1283,9 @@ void URL::Parse(const char* input,
         }
         break;
       case kNoScheme:
-        cannot_be_base = base->flags & URL_FLAGS_CANNOT_BE_BASE;
+        if (has_base) {
```

Comment:
I'll split it out as it's a bug in a code path actually used by code in master right now.